### ISSUE TITLE
JSON-LD verifiable credential with multiple subjects

### ISSUE THREAD

**Issue by awoie** [w3c/vc-data-model#518]
> A verifiable credential contains claims about one or more subjects.

According to the current spec the data model contains a `credentialSubject` which is:
> The value of the credentialSubject property is defined as a set of objects that contain one or more properties that are each related to a subject of the verifiable credential. Each object MAY contain an id as described in Section § 4.2 Identifiers.

Could you please provide an example of a verifiable credential containing multiple subjects.

**Comment by David-Chadwick** [comment]
```
...
"type": ["VerifiableCredential", "MarriageCertificate"],
  "credentialSubject": [{
    "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
    "marriedTo": "did:example:ebc276e12ec21feb1f712ebc6f1"
  },{
    "id": "did:example:ebc276e12ec21feb1f712ebc6f1",
    "marriedTo": "did:example:ebfeb1f712ebc6f1c276e12ec21"
  }],
....
```

**Comment by awoie** [comment]
@David-Chadwick Thank you. In that case I have still another question.

The VC spec contains the following example:
```
{
  ...
  "credentialSubject": {
    "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
    "degree": {
      "type": "BachelorDegree",
      "name": "<span lang='fr-CA'>Baccalauréat en musiques numériques</span>"
    }
  },
  ...
}
```
The `credentialSubject` is defined as a:
> set of objects that contain one or more properties

Note that in JSON a set of properties is mapped onto `Object type`.

The example above is not an **array** of properties either. Is the example wrong, or is this just JSON-LD magic?

**Comment by awoie** [comment]
@dlongley @msporny could you please clarify?

**Comment by msporny** [comment]
> @dlongley @msporny could you please clarify?

Yes. In JSON-LD, all properties can have one or more "objects" associated with them. So, 

```
"foo": { ... }
```

is effectively shorthand for:

```
"foo": [{ ... }]
```
 
So, all of the examples are correct.

In practice, one can use a `credentialSchema` to further restrict the JSON-LD and JSON forms so that you force developers to always use array form, for example.

In the spec, the data model states "a set of objects". A serialization can choose to use the shorthand if there is only one object (and then enforce that there is always only one object by using `credentialSchema`).

Does this answer your question @awoie? If so, can we close this issue?

**Comment by awoie** [comment]
@msporny That is what I thought. I am still not sure how a plain JSON representation of multiple `credentialSubject` instances looks like. It might be even a bit unclear how a single `credentialSubject` is represented in plain JSON.

The spec reads:
> The value of the credentialSubject property is defined as a set of objects that contain one or more properties ...

JSON-LD maps onto JSON as follows according to the spec:
> Unordered sets of values SHOULD be represented as an Array type.
> Sets of properties SHOULD be represented as an Object type.

If 1. is applicable, then the non-normative example would be different in JSON as it will ALWAYS be an array. If the JSON representation is either an array or a single set of objects, then JSON processors would need additional processing rules. 

```
"credentialSubject": {
    "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
    "degree": {
      "type": "BachelorDegree",
      "name": "<span lang='fr-CA'>Baccalauréat en musiques numériques</span>"
    }
  },
```

If 2. is applicable, then I don't know how you would represent multiple `credentialSubject` instances in JSON. 

Either way, we should provide a non-normative example on how to define multiple `credentialSubject` instances in any case for clarification purposes.

**Comment by msporny** [comment]
Add example showing multiple credential subjects to the specification.

**Comment by burnburn** [comment]
WG resolution on 28 May 2019 call:

RESOLUTION: Issue #518 is resolved via PR #644, which adds an example on multiple subjects to the specification. Close issue #518 after the PR has been merged.

**Comment by burnburn** [comment]
PR merged, closing.
